nidx/src/metadata/shard.rs

```diff
@@ -18,27 +18,43 @@
 // along with this program. If not, see <http://www.gnu.org/licenses/>.
 //
 use super::index::*;
-use sqlx::{Executor, Postgres};
+use sqlx::{types::time::PrimitiveDateTime, Executor, Postgres};
 use uuid::Uuid;
 
 pub struct Shard {
     pub id: Uuid,
     pub kbid: Uuid,
+    pub deleted_at: Option<PrimitiveDateTime>,
 }
 
 impl Shard {
-    pub async fn create(meta: impl Executor<'_, Database = Postgres>, kbid: Uuid) -> Result<Shard, sqlx::Error> {
+    pub async fn create(meta: impl Executor<'_, Database = Postgres>, kbid: Uuid) -> sqlx::Result<Shard> {
         sqlx::query_as!(Shard, "INSERT INTO shards (kbid) VALUES ($1) RETURNING *", kbid).fetch_one(meta).await
     }
 
-    pub async fn get(meta: impl Executor<'_, Database = Postgres>, id: Uuid) -> Result<Shard, sqlx::Error> {
+    pub async fn get(meta: impl Executor<'_, Database = Postgres>, id: Uuid) -> sqlx::Result<Shard> {
         sqlx::query_as!(Shard, "SELECT * FROM shards WHERE id = $1", id).fetch_one(meta).await
     }
 
+    pub async fn try_get(meta: impl Executor<'_, Database = Postgres>, id: Uuid) -> sqlx::Result<Option<Shard>> {
```

Comment:
Consider removing this and instead, call `get` and match the result against `Err(RowNotFound)`